### ISSUE TITLE
Bug in Nameserver.pm line 301

### ISSUE THREAD

**Issue by matsduf** [dotse/zonemaster-engine#384]
At times 1.24, 1.36 and 11.59 Nameserver.pm spits out errors that indicates that it makes wrong assumptions about data.

```
$ zonemaster-cli tplinkdns.com --level DEBUG
Seconds Level     Message
======= ========= =======
   0.00 DEBUG     SYSTEM:START_TIME string=2018-04-05 15:57:54 +0200; time_t=1522936674
   0.00 DEBUG     SYSTEM:TEST_TARGET module=all; zone=tplinkdns.com
   0.00 DEBUG     Using module Zonemaster::Engine::Test::Basic version v1.0.7.
   0.00 INFO      Using version v2.0.6 of the Zonemaster engine.
   0.01 DEBUG     Using prerequisite module Zonemaster::LDNS version 1.0.2.
   0.01 DEBUG     Using prerequisite module IO::Socket::INET6 version 2.72.
   0.01 DEBUG     Using prerequisite module Moose version 2.1604.
   0.01 DEBUG     Using prerequisite module Module::Find version 0.13.
   0.01 DEBUG     Using prerequisite module JSON version {version}.
   0.01 DEBUG     Using prerequisite module File::ShareDir version 1.102.
   0.01 DEBUG     Using prerequisite module File::Slurp version 9999.19.
   0.02 DEBUG     Using prerequisite module Net::IP version 1.26.
   0.02 DEBUG     Using prerequisite module List::MoreUtils version 0.413.
   0.02 DEBUG     Using prerequisite module Mail::RFC822::Address version 0.3.
   0.02 DEBUG     Using prerequisite module Scalar::Util version 1.41.
   0.02 DEBUG     Using prerequisite module Hash::Merge version 0.200.
   0.02 DEBUG     Using prerequisite module Readonly version 2.
   0.02 INFO      Configuration was read from /usr/local/share/perl/5.22.1/auto/share/dist/Zonemaster-Engine/config.json.
   0.03 INFO      Policy was read from /usr/local/share/perl/5.22.1/auto/share/dist/Zonemaster-Engine/policy.json.
   0.28 INFO      Parent domain 'com' was found for the tested domain.
   1.24 DEBUG     DNS query to ns2.tplinkdns.com/54.87.217.253 for ns1.tplinkdns.com/AAAA/IN failed with error: answer section incomplete at /usr/local/share/perl/5.22.1/Zonemaster/Engine/Nameserver.pm line 301, <DATA> line 20.
   1.36 DEBUG     DNS query to ns1.tplinkdns.com/52.204.177.89 for ns1.tplinkdns.com/AAAA/IN failed with error: answer section incomplete at /usr/local/share/perl/5.22.1/Zonemaster/Engine/Nameserver.pm line 301, <DATA> line 20.
   1.46 DEBUG     SYSTEM:IS_BLACKLISTED class=IN; dnssec=0; message=Server transport has been blacklisted due to previous failure; name=ns2.tplinkdns.com; ns=ns2.tplinkdns.com/54.87.217.253; proto=UDP; type=A
   1.47 DEBUG     SYSTEM:IS_BLACKLISTED class=IN; dnssec=0; message=Server transport has been blacklisted due to previous failure; name=ns2.tplinkdns.com; ns=ns1.tplinkdns.com/52.204.177.89; proto=UDP; type=A
   1.57 DEBUG     SYSTEM:IS_BLACKLISTED class=IN; dnssec=0; message=Server transport has been blacklisted due to previous failure; name=ns2.tplinkdns.com; ns=ns2.tplinkdns.com/54.87.217.253; proto=UDP; type=AAAA
   1.57 DEBUG     SYSTEM:IS_BLACKLISTED class=IN; dnssec=0; message=Server transport has been blacklisted due to previous failure; name=ns2.tplinkdns.com; ns=ns1.tplinkdns.com/52.204.177.89; proto=UDP; type=AAAA
   1.58 INFO      IPv4 is enabled, can send "NS" query to ns1.tplinkdns.com/80.217.70.196.
  11.59 DEBUG     DNS query to ns1.tplinkdns.com/80.217.70.196 for tplinkdns.com/NS/IN failed with error: Could not send or receive, because of network error at /usr/local/share/perl/5.22.1/Zonemaster/Engine/Nameserver.pm line 301, <DATA> line 20.
  11.59 NOTICE    Nameserver ns1.tplinkdns.com/80.217.70.196 did not respond to NS query.
  11.59 INFO      IPv4 is enabled, can send "A" query to ns1.tplinkdns.com/80.217.70.196.
  11.59 DEBUG     SYSTEM:IS_BLACKLISTED class=IN; dnssec=0; message=Server transport has been blacklisted due to previous failure; name=www.tplinkdns.com; ns=ns1.tplinkdns.com/80.217.70.196; proto=UDP; type=A
  11.59 INFO      Nameservers did not respond to A query.
  11.59 DEBUG     Module Zonemaster::Engine::Test::Basic finished running.
  11.59 CRITICAL  Not enough data about tplinkdns.com was found to be able to run tests.
```

From Nameserver.pm line 301:

```
   300      else {
   301          $res = eval { $self->dns->query( "$name", $type, $href->{class} ) };
   302          if ( $@ ) {
   303              my $msg = "$@";
   304              chomp( $msg );
   305              Zonemaster::Engine->logger->add( LOOKUP_ERROR =>
   306                    { message => $msg, ns => "$self", name => "$name", type => $type, class => $href->{class} } );
   307              $self->blacklisted->{ $flags{usevc} }{ $flags{dnssec} } = 1;
   308              if ( !$flags{dnssec} ) {
   309                  $self->blacklisted->{ $flags{usevc} }{ !$flags{dnssec} } = 1;
   310              }
   311          }
```

**Comment by mattias-p** [comment]
We don't understand what's happening here, but in any case it's not reasonable to include source file references in the output.

**Comment by sandoche2k** [comment]
@matsduf close?

**Comment by matsduf** [comment]
Issue solved by #389.
